metadata/metadata.go

```diff
@@ -273,12 +273,26 @@ func (c *Client) updateEtag(resp *http.Response) bool {
 	return c.etag != oldEtag
 }
 
+func shouldRetry(resp *http.Response, err error) bool {
+	// If the context was canceled just return the error and don't retry.
+	if err != nil && errors.Is(err, context.Canceled) {
+		return false
+	}
+
+	// Known non-retriable status codes.
+	if resp != nil && resp.StatusCode == 404 {
+		return false
+	}
+
+	return true
+}
+
 func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
 	for i := 1; i <= backoffAttempts; i++ {
 		resp, err := c.do(ctx, cfg)
 
-		// If the context was canceled just return the error and don't retry.
-		if err != nil && errors.Is(err, context.Canceled) {
+		// Check if error is retriable, if not just return the error and don't retry.
+		if err != nil && !shouldRetry(resp, err) {
```

Comment:
It should fix one of the issues I noticed, but additionally we want ScheduleJob() to not block the caller, it all should run in its own go routine - with this new non retry'able logic we'll see the symptom go away but we may stumble on it again later when the ShouldEnable() depends on a long lasting operation. We can address it in a follow up PR though.

Reply:
Considering the purpose of `ShouldEnable()` it should not be a long running operation to have a noticeable block. Updated to run as separate go routine and not block agent main.